pkg/apprepo/apprepos_handler.go

```diff
@@ -125,13 +148,17 @@ func NewAppRepositoriesHandler(kubeappsNamespace string) (*AppRepositoriesHandle
 	if err != nil {
 		return nil, err
 	}
+	svcAppRepoClient, err := apprepoclientset.NewForConfig(svcRestConfig)
+	if err != nil {
+		return nil, err
+	}
 
-	return &AppRepositoriesHandler{
+	return &appRepositoriesHandler{
 		config:            *config,
 		kubeappsNamespace: kubeappsNamespace,
 		// See comment in the struct defn above.
 		clientsetForConfig: clientsetForConfig,
-		svcKubeClient:      svcKubeClient,
+		svcClientset:       &combinedClientset{svcAppRepoClient, svcKubeClient},
```

Comment:
So `a.clientset` isn't yet defined (as no one has called `AsUser()` or `AsSVC()` yet), so if a call-site now called `a.CreateAppRepository(..)` they'd get a run-time panic, I think? This is what I meant above about mixing responsibilities. Perhaps what is returned here should *not* implement the interface, and only the handler returned by `AsUser()` or `AsSVC()` should implement the interface. So the object returned here wouldn't be a handler, but enables you to create a handler.

Reply:
agree